pkgs/build-support/fetchgithub/default.nix

```diff
@@ -11,11 +14,16 @@ lib.makeOverridable (
 , ... # For hash agility
 }@args:
 
+assert (lib.assertMsg (lib.xor (tag == null) (rev == null)) "fetchFromGitHub requires either `rev` xor `tag`.");
+
 let
 
   position = (if args.meta.description or null != null
     then builtins.unsafeGetAttrPos "description" args.meta
-    else builtins.unsafeGetAttrPos "rev" args
+    else if tag != null then
+      builtins.unsafeGetAttrPos "tag" args
+    else
+      builtins.unsafeGetAttrPos "rev" args
```

Comment:
This is tricky and excellent, nice work.

Reply:
Thanks :)

fetchgit doesn't have this btw; we should probably fix this.

Reply:
Separate PR; this one is going in!